.github/workflows/pr-to-update-winget.yml

```diff
@@ -0,0 +1,129 @@
+          git config --global user.email "$user@users.noreply.github.com"
+
+      - name: Create node.js environment
+        uses: actions/setup-node@v4
+        with:
+          node-version: "21"
+
+      - name: Set latest tag
+        run: |
+          echo "Setting latest tag"
+          latestTag=$(git tag --list --sort=version:refname 'v*' | tail -1)
+          # Remove the 'v' from the tag
+          latestTag=${latestTag#v}
+          echo "LATEST_HEADLAMP_TAG=$latestTag" >> $GITHUB_ENV
+          echo $latestTag
+
+      - name: Install winget dependencies
+        run: |
+          cd $GITHUB_WORKSPACE/app/windows/winget
+          npm install
+
+      - name: Run winget script
+        run: |
+          echo "Running winget script"
+          echo "Repository: ${{ github.repository }}"
+          echo "Workspace: ${GITHUB_WORKSPACE}"
+          echo $GITHUB_WORKSPACE
+          pwd
+          echo "LATEST_HEADLAMP_TAG: $LATEST_HEADLAMP_TAG"
+          echo "LATEST_HEADLAMP_TAG: ${LATEST_HEADLAMP_TAG}"
+          echo ${LATEST_HEADLAMP_TAG}
+          cd $GITHUB_WORKSPACE/app/windows/winget
+          node winget-create.js $LATEST_HEADLAMP_TAG $GITHUB_WORKSPACE/app/windows/winget
+
+      - name: Print script finished
+        run: |
+          echo "Script finished"
+          pwd
+          cd $GITHUB_WORKSPACE/app/windows/winget
+          ls
+
+      - name: Checkout winget-pkgs
+        uses: actions/checkout@v4
+        with:
+          repository: headlamp-k8s/winget-pkgs
+          path: winget-pkgs
+          token: ${{ secrets.KINVOLK_REPOS_TOKEN }}
+          fetch-depth: 0
+
+      - name: Copy winget files
+        run: |
+          echo "Copying winget files"
+          echo "Repository: ${{ github.repository }}"
+          echo "Workspace: ${GITHUB_WORKSPACE}"
+          pwd
+          cd $GITHUB_WORKSPACE/app/windows/winget
+          ls
+          cd $GITHUB_WORKSPACE/app/windows/winget/$LATEST_HEADLAMP_TAG
+          pwd
+          ls
+          cp -r $GITHUB_WORKSPACE/app/windows/winget/$LATEST_HEADLAMP_TAG $GITHUB_WORKSPACE/winget-pkgs/manifests/h/Headlamp/Headlamp
+
+      - name: Create PR branch
+        run: |
+          user=${{github.actor}}
+          if [ -z $user ]; then
+            user=vyncent-t
+          fi
+          echo "Creating PR branch"
+          echo "Repository: ${{ github.repository }}"
+          echo "Workspace: ${GITHUB_WORKSPACE}"
+          pwd
+          echo "moving to winget-pkgs directory"
+          cd $GITHUB_WORKSPACE/winget-pkgs
+          pwd
+          ls
+          echo "moving to Headlamp directory"
+          cd $GITHUB_WORKSPACE/winget-pkgs/manifests/h/Headlamp/Headlamp
+          pwd
+          ls
+          git checkout -b "winget-update-$LATEST_HEADLAMP_TAG"
+          git add .
+          git commit -m "Update winget package $LATEST_HEADLAMP_TAG"
+          git commit -s --amend --no-edit
+          git push origin "winget-update-$LATEST_HEADLAMP_TAG"
```

Comment:
You can avoid pushing the branch directly if you use gh pr create.

Reply:
While using peter/evans or the create pr command, we run into the sso auth error from earlier- as fork it seems it cannot be reliably separated from the windows source repo we do not own. 

The command fails as the repo is being set to the current repo where the headlamp workflow originates and there is not a way to change that it seems.

By using the API we are able to post the pr directly to our winget repo fork reliably